libethereum/EthereumPeer.cpp

```diff
@@ -364,8 +372,15 @@ bool EthereumPeer::interpret(unsigned _id, RLP const& _r)
 	case NewBlockHashesPacket:
 	{
 		unsigned itemCount = _r.itemCount();
+
 		clog(NetMessageSummary) << "BlockHashes (" << dec << itemCount << "entries)" << (itemCount ? "" : ": NoMoreHashes");
 
+		if (itemCount > c_maxIncomingNewHashes)
```

Comment:
there's not technically a limit on this in the protocol. we should place a limit (e.g. 32) and make other clients aware of it.